### PR TITLE
INC-289: Create aggregate tables from Analytical Platform source data

### PR DESCRIPTION
**Pull request by ushkarev**

### CODE REVIEW

**Review by xoen — APPROVED** [review]

File: server/views/components/dataSource.njk [hunk 1]
```diff
@@ -2,6 +2,6 @@
   <p>
     Data source: {{ source -}}<span class="govuk-visually-hidden">.</span>
     <br />
-    Date updated: {{ date | date -}}<span class="govuk-visually-hidden">.</span>
+    Date updated: {{ date | shortDate -}}<span class="govuk-visually-hidden">.</span>
```
[the remark this answers is not in the document]
Comment (xoen) [reply server/views/components/dataSource.njk:5]: (thanks for clarifying this, it's to pause when using a screen reader 👌)

File: server/services/analyticsService.ts [hunk 2]
```diff
@@ -123,77 +134,68 @@ export default class AnalyticsService {
...
-      ['5', 13, 2],
-      ['6', 5, 14],
-      ['7', 12, 33],
-      ['H', 0, 0],
-      ['SEG', 0, 8],
-    ]
+    type AggregateRow = [string, number, number]
+    const aggregateTable = this.mapRowsAndSumTotals<StitchedRow, AggregateRow>(
+      filteredTables,
+      ([, wing, positives, negatives]) => [wing, positives, negatives],
```
[the remark this answers is not in the document]
Comment (ushkarev) [reply server/services/analyticsService.ts]: damn i missed this one

File: server/services/analyticsService.ts [hunk 3]
```diff
@@ -304,3 +306,20 @@ export default class AnalyticsService {
...
+  }
+  if (location2 === 'All') {
+    return 1
+  }
+  if (location1.length === 1 && location2.length !== 1) {
+    return -1
+  }
+  if (location1.length !== 1 && location2.length === 1) {
+    return 1
+  }
```
[the remark this answers is not in the document]
Comment (ushkarev) [reply server/services/analyticsService.ts:249]: yes: locations of length 1 go first, longer ones are considered special in some way so go at the end

File: server/views/components/dataSource.njk (hunk 1 above)
[the remark this answers is not in the document]
Comment (ushkarev) [reply server/views/components/dataSource.njk:5]: yup, because it's a single `p` element it'd be read as one sentence otherwise

**Pull request merged**